File: C/pub1/Distrib/LIB/LIB-PHP/Example/subp.echo.bat.run.php
#!php.exe
<?php
/* ********************************************************
 * subp.echo.bat.run.php
 * This Script will Run a echo-hello.bat Tool
**********************************************************/

# Setup Command Interpreter
$pathCMD = $_SERVER['SystemRoot'] . "\\System32";
$CMDEXE = $pathCMD . "\\" . "cmd.exe";
$batFile = "echo-hello.bat";
# $aStartCmd = "Start /b ";
$aStartCmd = $CMDEXE. " /k ";
$aStartPath = $_SERVER['USERPROFILE'];
$aBatPath = $aStartPath . "\\" . $batFile;
if( is_dir($pathCMD)) {
    if(is_file($CMDEXE)){
        $strValue = $aStartCmd . $aBatPath;
        if(is_file($aBatPath)) {
            $handle = (popen($strValue, 'r'));
            echo( "\n" . '$handle:' . gettype($handle) . "\n");
            while(!feof($handle)){
                $read = fread($handle, 4096);
                echo($read);
            }
            $r = pclose($handle);
            $rjson = json_encode(array('result'=>$r));
            print_r($rjson);
        }
    }
    else {
        echo("\nA File " . $CMDEXE . " is not found.");
    }
}
else {
    echo("\nA Path " . $pathCMD . " is not Found.");
}
?>

File: C-Source/pub1/Distrib/LIB/LIB-RUBY/Example/subp.echo.bat.run.rb
#!ruby.exe
#* ********************************************************
#* subp.echo.bat.run.php
#* This Script will Run a echo-hello.bat Tool
#*********************************************************/

require 'open3'

# Setup Command Interpreter
pathCMD = ENV['SystemRoot'] + "\\System32"
cmdEXE = pathCMD + "\\" + "cmd.exe"
aBatFile = "echo-hello.bat"
# aStartCmd = "Start /b "
aStartCmd = cmdEXE + " /c "
aStartPath = ENV['USERPROFILE']
aBatPath = aStartPath + "\\" + aBatFile
if File.directory?(pathCMD) then
    if File.file?(cmdEXE) then
        strValue = aStartCmd + aBatPath
        # puts aBatPath
        if File.file?(aBatPath) then
            stdout,stderr,status=Open3.capture3(strValue)
            puts "\nSTDOUT=\n"
            puts stdout
            puts "\nSTDERR=\n"
            puts stderr
            puts "\nResult: #{status.exitstatus}"
        else
            puts "\nA File #aPingExe is not Found."
        end
    else
        puts "\nA File #cmdExe is not Found."
    end
else
    puts "\nA Folder #pathCMD is not Found."
end